Mathlib/RingTheory/HopfAlgebra.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Ali Ramsey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ali Ramsey
+-/
+import Mathlib.RingTheory.Bialgebra
+
+/-!
+# Hopf algebras
+
+In this file we define `HopfAlgebra`, and provide instances for:
+
+* Commutative semirings: `CommSemiring.toHopfAlgebra`
+
+# Main definitions
+
+* `HopfAlgebra R A` : the Hopf algebra structure on an `R`-bialgebra `A`.
+* `HopfAlgebra.antipode R A` : The `R`-linear map `A →ₗ[R] A`
+
+## TODO
+
+* Uniqueness of Hopf algebra structure on a bialgebra (i.e. if the algebra and coalgebra structures
+agree then the antipodes must also agree).
+
+* S(1)=1 and S(ab)=S(b)S(a), so in particular if A is commutative then S is an algebra homomorphism.
```

Comment:
```suggestion
* $S(1)=1$ and $S(ab)=S(b)S(a)$, so in particular if A is commutative then S is an algebra homomorphism.
```
or
```suggestion
* `S 1 = 1` and `S (a * b) = S b * S a`, so in particular if A is commutative then S is an algebra homomorphism.
```
so that this looks nice in the docs.